File: 01.py
#!/usr/bin/env python3
from sys import stdin

ls   = [int(l.strip()) for l in stdin]
cnt  = sum([a < b for a,b in zip(ls[:-1], ls[1:])])
ws   = [a+b+c for a,b,c in zip(ls[:-2], ls[1:-1], ls[2:])]
cnt2 = sum([a < b for a,b in zip(ws[:-1], ws[1:])])

print (f"Part 1: {cnt}, Part 2: {cnt2}")

File: 01.scala
#!/usr/bin/env scala

import scala.io.Source.stdin
import scala.util.chaining._

@main def aoc(): Unit = 
  val ns = stdin.getLines().map(_.toInt).toList
  val g = (a:Int, b:Int)  => if b > a then 1 else 0
  val f = (xs:List[Int]) => xs.pipe(ns=>ns.zip(ns.drop(1))).map(g.tupled).sum

  println(s"Part 1: ${ f(ns) }, Part 2: ${ f(ns.sliding(3).map(_.sum).toList) }")
